--- /home/cons/EMPHASIS-Layer/cenzontle_git_projects/graphql-server-model-codegen/docker/integration_test_run/instance1/validations/accession.js	2020-05-25 12:42:35.233146658 +0200
+++ accession_validation.js	2020-05-25 12:53:59.505370697 +0200
@@ -15,7 +15,8 @@
                 "type": ["string", "null"]
             },
             "collectors_name": {
-                "type": ["string", "null"]
+                "type": "string",
+                "pattern": "^[a-zA-Z0-9_]+$"
             },
             "collectors_initials": {
                 "type": ["string", "null"]
@@ -66,4 +67,4 @@
     }
 
     return accession
-}
\ No newline at end of file
+}
