--- individual.js.orig	2020-06-29 18:51:36.275316065 -0500
+++ individual.js	2020-06-29 18:51:45.375659525 -0500
@@ -4,6 +4,8 @@
 const ajv = validatorUtil.addDateTimeAjvKeywords(new Ajv({
     allErrors: true
 }))
+const path = require('path');
+const individual_model =  require(path.join(__dirname, '..', 'models', 'sql' ,'individual.js'));
 
 // Dear user, edit the schema to adjust it to your model
 module.exports.validator_patch = function(individual) {
@@ -11,7 +13,7 @@
     individual.prototype.validationControl = {
         validateForCreate: true,
         validateForUpdate: true,
-        validateForDelete: false,
+        validateForDelete: true,
         validateAfterRead: false
     }
 
@@ -19,7 +21,15 @@
         "$async": true,
         "properties": {
             "name": {
-                "type": ["string", "null"]
+                "anyOf": [
+                  {
+                    "type": "null"
+                  },
+                  {
+                    "type": "string",
+                    "pattern": "^[a-zA-Z0-9]+$"
+                  }
+                ]
             }
         }
     }
@@ -41,12 +51,9 @@
 
     individual.prototype.validateForDelete = async function(id) {
 
-        //TODO: on the input you have the id of the record to be deleted, no generic
-        // validation checks are available. You might need to import the correspondant model
-        // in order to read the whole record info and the do the validation.
-
-        return {
-            error: null
+      let record = await individual_model.readById(id);
+        if (record.name === "Undeletable") {
+          throw new Ajv.ValidationError('Record with name "Undeletable" can\'t be deleted')
         }
     }
 
