--- aminoacidsequence.js	2020-10-06 18:33:07.223334159 -0500
+++ aminoacidsequence_update.js	2020-10-06 18:34:57.729008780 -0500
@@ -75,10 +75,9 @@
      */
     static async readById(id, benignErrorReporter) {
 
-        /*
-        YOUR CODE GOES HERE
-         */
-        throw new Error('readById() is not implemented for model aminoacidsequence');
+      let res = request('get', baseUrl + id);
+      if(res.statusCode !== 200) throw new Error(`Remote service response status code for ${id}: ${res.statusCode}`);
+      return new aminoacidsequence(JSON.parse(res.body.toString('utf8')));
     }
 
     /**
@@ -137,10 +136,22 @@
      */
     static async readAll(search, order, pagination, benignErrorReporter) {
 
-        /*
-        YOUR CODE GOES HERE
-        */
-        throw new Error('readAll() is not implemented for model aminoacidsequence');
+      let ids = [69905, 10644, 63165];
+      let unfiltered = [];
+      let cnt = 0;
+      for(let idx in ids){
+        if(idx >= pagination.offset) {
+          console.log(ids[idx]);
+          let res = request('get', baseUrl + ids[idx]);
+          if (res.statusCode !== 200) throw new Error(`Remote service response status code for ${ids[idx]}: ${res.statusCode}`);
+          unfiltered.push(new aminoacidsequence(JSON.parse(res.body.toString('utf8'))));
+          cnt++;
+          if(cnt === pagination.limit)
+          break;
+        }
+      }
+
+      return unfiltered;
     }
 
     /**
